heareval/tasks/pipeline.py

```diff
@@ -286,37 +271,65 @@ def split_train_test_val(self, metadata: pd.DataFrame):
             + f"now sampled with split key are: {splits_to_sample}"
         )
 
-        # Depending on whether valid and test are already present, the percentage can
-        # either be the predefined percentage or 0
-        validation_percentage = (
-            VALIDATION_PERCENTAGE if "valid" in splits_to_sample else 0
-        )
-        test_percentage = TEST_PERCENTAGE if "test" in splits_to_sample else 0
-
-        metadata.reset_index(drop=True, inplace=True)
-        metadata[metadata["split"] == "train"] = metadata[
-            metadata["split"] == "train"
-        ].assign(
-            split=lambda df: df["split_key"].apply(
-                # Use the which set to split the train into the required splits
-                lambda split_key: which_set(
-                    split_key,
-                    validation_percentage=validation_percentage,
-                    test_percentage=test_percentage,
-                )
+        train_percentage: float
+        valid_percentage: float
+        test_percentage: float
+
+        # If we want a 60/20/20 split, but we already have test
+        # then we want to do a 75/25/0 split so that train is still 3x validation
+        if splits_to_sample == set():
+            return metadata
+        elif splits_to_sample == set("valid"):
+            tot = TRAIN_PERCENTAGE + TEST_PERCENTAGE
```

Comment:
The tot whether your sampling just "valid" or just "test" is the same -- is this not supposed to be TRAIN_PERCENTAGE + VALIDATION_PERCENTAGE?

Reply:
Yeah look now